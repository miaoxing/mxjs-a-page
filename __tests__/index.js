import React from 'react';
import {Page, PageContext} from '..';
import {render} from '@testing-library/react';
import {setUrl, resetUrl} from '@mxjs/test';

describe('Page', () => {
  test('basic', () => {
    const result = render(<Page/>);
    expect(result.container).toMatchSnapshot();
  });

  test('breadcrumb ignore empty path name', () => {
    setUrl('/admin/test/child');

    const result = render(<PageContext.Provider value={{
      pages: {
        '/admin': {
          name: 'Home',
          '/test': {
            '/child': {
              name: 'Child Page',
            },
          },
        },
      },
    }}>
      <Page/>
    </PageContext.Provider>);

    expect(result.container).toMatchSnapshot();

    resetUrl();
  });
});

