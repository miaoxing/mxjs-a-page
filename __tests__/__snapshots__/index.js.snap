// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page basic 1`] = `
<div>
  <div
    class="css-156ub72"
  >
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item css-1ch7sjt-invisible"
        >
          <a
            href="#"
            role="button"
          >
            ...
          </a>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="css-o1za7s"
  />
</div>
`;

exports[`Page breadcrumb ignore empty path name 1`] = `
<div>
  <div
    class="css-156ub72"
  >
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            href="#"
            role="button"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
        >
          Child Page
        </li>
        <li
          class="breadcrumb-item css-1ch7sjt-invisible"
        >
          <a
            href="#"
            role="button"
          >
            ...
          </a>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="css-o1za7s"
  />
</div>
`;
